various improvements to helm chart

draeron opened the pull request:
- added configuration of LDAP
- added log level configuration
- added service annotations
- added storage class specification
- fix bad registry configmap : https://github.com/kubic-project/caasp-services/issues/37

stefsuse commented:
This is awesome, thank you for looking into this, especially ConfigMap! 

Since this is a larger change and there were some PRs before you I would like to get some of those merged before I do a full review. 

One note for now - persistent storage is disabled by default in community charts unless it is absolutely necessary for the app to function.

draeron commented:
i've reviewed and reverted some default settings i changed by error (the persistence setting was one of them).

stefsuse commented on contrib/helm-charts/portus/values.yaml:
```diff
@@ -146,13 +216,18 @@ nginx:
 
+    ## Annotation to be added to the service
+    ##
+    # annotations:
```
uncomment?

stefsuse commented on contrib/helm-charts/portus/values.yaml:
```diff
@@ -53,11 +62,60 @@ portus:
+    ## online docs: http://port.us.org/features/2_LDAP-support.html.
+    ldap:
+      enabled: false
+
```
remove newline

stefsuse reviewed (changes requested):
Overall looks great. I would like to see NOTES.txt updated to include LDAP instructions when applicable, but I won't hold up changes over it.

Please rebase and increment version.

amaari-stingray committed:
added ldap parameters

- added ability for extra annotation on nginx service
- enabled redeployment on config map change
- added deployment log level
- moved ldap password to secret
- fixed configmap bad port printing
- version bump

amaari-stingray committed:
review correction

draeron commented:
i've made both correction, rebased and squashed the commits

i wouldn't mind adding a section in `notes.txt` but honestly i'm not sure what should be said about ldap in there as the usage ain't much different. I guess replacing the _Create admin user account_ with alternate text saying the first login will be promoted to admin account ?

stefsuse commented:
notes.txt should include basic information of what was deployed and how to get started (it does now) and important information that the chart can't do for you automatically. Some charts will also give a report of any add-ons you enabled. 

Like I said it was just a suggestion if anything came to mind that could help someone get started using the feature. I guess it is normally implied that the first user created is an admin.

stefsuse reviewed (approved):
LGTM and tests passed (sorry for delay on CI)
